src/data/markdown/docs/07 extensions/01 Guides/02 bundling.md

```diff
@@ -3,63 +3,112 @@ title: 'Build a k6 binary with extensions'
+$ xk6 build latest \
+  --with github.com/grafana/xk6-sql@v0.0.1 \
   --with github.com/grafana/xk6-output-prometheus-remote
 ```
 
-This will build a `k6` binary in the current directory based on the most recently
-released k6 version including the [`xk6-sql`](https://github.com/grafana/xk6-sql) and [`xk6-output-prometheus-remote`](https://github.com/grafana/xk6-output-prometheus-remote). 
+> When creating your bundles, use the [interactive builder](/extensions/bundle-builder/) to avoid typing mistakes!
+
+Once completed, the **current directory** will contain your newly built `k6` binary with 
+the [`xk6-sql`](https://github.com/grafana/xk6-sql) and [`xk6-output-prometheus-remote`](https://github.com/grafana/xk6-output-prometheus-remote) 
+extensions.
 
 ```bash
 ... [INFO] Build environment ready
 ... [INFO] Building k6
 ... [INFO] Build complete: ./k6
+
+xk6 has now produced a new k6 binary which may be different than the command on your system path!
+Be sure to run './k6 run <SCRIPT_NAME>' from the '...' directory.
+```
+
+## Breaking down the xk6 command
+
+From the [xk6 documentation](https://github.com/grafana/xk6/#command-usage), the command line usage is as follows:
+
+```plain
+xk6 build [<k6_version>]
+    [--output <file>]
+    [--with <module[@version][=replacement]>...]
+    [--replace <module=replacement>...]
+    
+Flags:
+  --output   specifies the new binary name [default: 'k6']
+  --replace  enables override of dependencies for k6 and extensions [default: none]
+  --with     the extension module to be included in the binary [default: none]
 ```
 
-Now you can run a script with this binary and use the functionalities of the bundled extensions.
+> The use of `--replace` should be considered an advanced feature to be avoided unless required.
+
+Referring back to our executed command, we'll see that:
+- We specify the version as `latest`, which is also the default if we hadn't supplied 
+  a version. `latest` means that we'll build using the _latest_ source code for k6. Consider using 
+  a stable [release version](https://github.com/grafana/k6/releases) as a best practice unless
+  you genuinely want the _bleeding edge_.
+- With each `--with`, we specified a full GitHub URI for the extension repository. If not specifying 
+  a version, the default is `latest` once again. Check your extension repository for stable
+  release versions, if available, to lock in your version as we've done with `xk6-sql@v0.0.1`.
+- We did not use the `--output` option; therefore, our new binary is `k6` within the current directory. 
+  Had we used `--output k6-extended`, our binary would be named `k6-extended` within the current 
+  directory. If a directory is specified, then the new binary would be `k6` within 
+  _that_ directory. If a path to a non-existent file, e.g. `/tmp/k6-extended`, this will be the
+  path and filename for the binary.
+
+> Suppose now you've cloned the `xk6-sql` repository and want the local version included in your 
```

Comment:
This is a frequent case when building or debugging extensions.

I'd create an H3 or H2 section so it is highlighted on the Table Of Content (right sidebar).  The section could be something like `Building an extension from a local repository`